youtube_dl/__init__.py

```diff
@@ -169,7 +169,7 @@ def _real_main(argv=None):
         if not opts.audioquality.isdigit():
             parser.error('invalid audio quality specified')
     if opts.recodevideo is not None:
-        if opts.recodevideo not in ['mp4', 'flv', 'webm', 'ogg', 'mkv']:
+        if opts.recodevideo not in ['mp4', 'flv', 'webm', 'ogg', 'mkv', 'xvid']:
```

Comment:
By the way @dstftw, mkv is also another generic container, this does not reinforce my argument? As XviD is expected to be inside a AVI, is not this a better approach?

Well this is my proposal, but i accept to change the `opts.recodevideo` value to `avi` if this is your final decision. No problem.